indico/modules/events/models/reviews.py

```diff
@@ -19,44 +19,54 @@
 from sqlalchemy.ext.hybrid import hybrid_property
 
 from indico.util.caching import memoize_request
-from indico.util.i18n import _
-from indico.util.struct.enum import RichIntEnum
+from indico.util.locators import locator_property
 from indico.web.flask.util import url_for
 
 
-class ProposalMixin(object):
-    proposal_type = None
-    call_for_proposals_attr = None
+class ProposalGroupProxy(object):
+    """Represents the object that the proposals can be grouped by.
 
-    # endpoints
-    delete_comment_endpoint = None
-    create_comment_endpoint = None
-    edit_comment_endpoint = None
-    create_review_endpoint = None
-    edit_review_endpoint = None
+    It provides all necessary methods for building the URLs, displaying the
+    grouping information, etc.
+    """
+
+    title_attr = 'title'
+    full_title_attr = 'full_title'
+
+    def __init__(self, group):
+        self.instance = group
+
+    def __eq__(self, other):
+        if not isinstance(other, ProposalGroupProxy):
+            return False
+        return self.instance == other.instance
+
+    def __hash__(self):
+        return hash(self.instance)
+
+    def __ne__(self, other):
+        return not (self == other)
 
     @property
-    def cfp(self):
-        return getattr(self.event_new, self.call_for_proposals_attr)
+    def title(self):
+        return getattr(self.instance, self.title_attr)
 
-    def can_comment(self, user):
-        raise NotImplementedError
+    @property
+    def full_title(self):
+        return getattr(self.instance, self.full_title_attr) if self.full_title_attr else self.title
 
-    def can_review(self, user, check_state=False):
-        raise NotImplementedError
+    @locator_property
+    def locator(self):
+        return self.instance.locator
 
-    def get_delete_comment_url(self, comment):
-        return url_for(self.delete_comment_endpoint, comment)
 
-    def get_save_comment_url(self, comment=None):
-        return (url_for(self.edit_comment_endpoint, comment)
-                if comment
-                else url_for(self.create_comment_endpoint, self))
+class ProposalRevisionMixin(object):
```

Comment:
I would add docstring to the class and maybe to the attributes as well (`#: ...`).